### ISSUE TITLE
Can't create blog articles in all languages...

### ISSUE THREAD

**Issue by jedie** [nephila/djangocms-blog#307]
I have a installation with en and de languages... But i can't write blog articles in de, because the content form is completely missing.

How to fix that?

-----

![blog_de](https://cloud.githubusercontent.com/assets/71315/17146044/24769726-535d-11e6-9a76-4c16affcbc07.PNG)

-----

![blog_en](https://cloud.githubusercontent.com/assets/71315/17146015/0308a4bc-535d-11e6-869c-1c9317fe6adf.PNG)

-----

**Comment by jedie** [comment]
Some version information:
* djangocms-blog==0.8.5
* django-cms==3.3.1

**Comment by yakky** [comment]
Does categories and apphook config have a german translation?

**Comment by jedie** [comment]
I didn't see how to translate a category:
![djangocmsblog1](https://cloud.githubusercontent.com/assets/71315/17208461/2283d462-54b9-11e6-8a29-0e905f4e6330.PNG)

Also the config has nothing with translation.

**Comment by yakky** [comment]
Is ``PARLER_LANGUAGES`` configured https://djangocms-blog.readthedocs.io/en/latest/installation.html#minimal-configuration ?

**Comment by jedie** [comment]
Ah! There is a wrong dict key number in my `PARLER_LANGUAGES` so the problematic SITE_ID used 'default' and that's only: ` 'default': {'code': 'de', 'fallbacks': ['de'], 'hide_untranslated': False}`

After i correct this, i see the language select tabs. But there are still some errors:

- /**en**/admin/djangocms_blog/post/add/?language=**en** -> All ok, i see the Abstract form field
- /**en**/admin/djangocms_blog/post/add/?language=**de** -> All ok, i see the Abstract form field
- /**de**/admin/djangocms_blog/post/add/?language=**en** -> Error: No Abstract form field
- /**de**/admin/djangocms_blog/post/add/?language=**de** -> Error: No Abstract form field

IMHO some good TODOs for djangocms-blog:
* add `list_display = [...]` to `BlogConfigAdmin` and display languages.

btw. 1) it's a little bit boring, that we have to setup `CMS_LANGUAGES` and `PARLER_LANGUAGES` with similar data... What's about a helper for a DRY settings.py ?

btw. 2) The language select tabs are some kind of dangerous, because they are GET request and doesn't save on leaving or warn. e.g.: Start creating a new blog article in english and just click on a other langauge: All types data are gone and not saved.

**Comment by yakky** [comment]
@jedie actually with newer parler version (1.6 maybe, can't remember from the top of my head) you can skip PARLER_LANGUAGES and it will be derived from CMS_LANGUAGES automatically. There are a couple of differences in keys that prevents "PARLER_LANGUAGES = CMS_LANGUAGES", and conversion is handled internally by parler.
For the language tabs, I have a few notes about improving them in parler, but not yet time to tackle that. Maybe opening an issue in django-parler?
You are right, category admin needs some love, now that #125 is mostly dead

**Comment by yakky** [comment]
@jedie for the "de" issue, check whether apphook config exists in german (from /admin/djangocms_blog/blogconfig/)

**Comment by jedie** [comment]
I translated the config, e.g:
* DE: App.title: "BlogDE" - Object Name: "Artikel"
* EN: App.title: "BlogEN" - Object Name: "Article"

Strange is the representation of the App config:

----

![unbenannt](https://cloud.githubusercontent.com/assets/71315/17216810/683b25cc-54e2-11e6-8ee4-82cbfbe3d3a9.PNG)

----

I would expected to see "BlogEN" here?!?



But now i have the Problem of redirect errors...
Some urls:
```
/en/admin/djangocms_blog/post/479/en/en/en/en/en/en/en/en/en/en/en/en/en/en/en/en/en/en/en/en/?app_config=1&language=en

/en/admin/djangocms_blog/blogcategory/en/en/en/en/en/en/en/en/en/en/en/en/en/en/en/en/en/en/en/en/

/de/blog/en/en/en/en/en/en/en/en/en/en/en/en/en/en/en/en/en/en/en/en/
```

**Comment by jedie** [comment]
For the dangerous language tabs staff exist already a issued here: https://github.com/django-parler/django-parler/issues/151

**Comment by yakky** [comment]
@jedie No, the blog config label in the admin list (and everywhere is used) does not use the "title" which is meant to be a public facing label.
Under which circumstances you get the redirection error?

**Comment by jedie** [comment]
> Under which circumstances you get the redirection error?

That was unrelated. Just a typo in .htaccess ;)


But now i get TypeError in admin and my pages are down in non-admin. Very Strange :(

Maybe a bad idea to use djangocms_multisite, too.

The Language related settings:
```
LANGUAGES = (
    ('de', _('German')),
    ('en', _('English')),
)
LANGUAGE_CODE = 'de'
CMS_LANGUAGES = {
    2: [
        {
            'code': 'en',
            'name': _('English'),
            'public': True,
            'hide_untranslated': False,
        },
        {
            'code': 'de',
            'name': _('Deutsch'),
            'public': True,
            'hide_untranslated': False,
        },
    ],
    3: [
        {
            'code': 'de',
            'name': _('Deutsch'),
            'public': True,
            'fallbacks': [],
        },
    ],
    4: [
        {
            'code': 'de',
            'name': _('Deutsch'),
            'public': True,
            'fallbacks': [],
        },
    ],
    'default': {
        'fallbacks': ['en', 'de'],
        'redirect_on_fallback': True,
        'public': True,
        'hide_untranslated': False,
    },
}

PARLER_DEFAULT_LANGUAGE_CODE = LANGUAGE_CODE
PARLER_LANGUAGES = {
    2: (
        {'code': 'en',},
        {'code': 'de',},
    ),
    3: (
        {'code': 'de',},
    ),
    4: (
        {'code': 'de',},
    ),
    'default': {
        'fallback': PARLER_DEFAULT_LANGUAGE_CODE,
        'hide_untranslated': False,
    }
}
```

**Comment by yakky** [comment]
could you post the traceback?

**Comment by jedie** [comment]
https://gist.github.com/jedie/c39e6ecfbacb1e2829bb9db24be8f9aa

The Traceback comes if **/de/** used, but not if **/en/** is used.
Don't know, maybe it's related to: https://github.com/django-parler/django-parler/issues/90

- /**de**/admin/djangocms_blog/blogcategory/add/ -> OK
- /**en**/admin/djangocms_blog/blogcategory/add/ -> Same error

e.g.: that works:
- /de/admin/djangocms_blog/blogconfig/1/
- /en/admin/djangocms_blog/blogconfig/1/

**Comment by jedie** [comment]
OK, my sites are back again... I restored a old version of my settings.py...
The pages are down, because i have misconfiguration the 'default' cache. That's all.

Back to these issues: Now I realized:

If DEBUG=True, then i get the Tracebacks with the **TypeError: __str__ returned non-string (type NoneType)**
If DEBUG=False, then i see only the missing form like:

![blog_de](https://cloud.githubusercontent.com/assets/71315/17146044/24769726-535d-11e6-9a76-4c16affcbc07.PNG)

----

So the real error, comes by rendering the blog category form. (Strange that django just hide the error if DEBUG=False and not raise a internal server error)

**Comment by yakky** [comment]
Could you provide settings.py file, virtualenv and a dump of blog config / categories data?
I can't reproduce this on both 0.8 and develop branches

**Comment by yakky** [comment]
@jedie it's this still the case?
